forms.fill.high.numbers.js

```diff
@@ -0,0 +1,174 @@
+// ==UserScript==
+// @name         Dark Galaxy - Fill forms with high numbers
+// @namespace    https://darkgalaxy.com/
+// @version      0.2
+// @description  All your planet are belong to us
+// @author       Riddick
+// @homepage     https://github.com/fl0v/dg
+// @supportURL   https://github.com/fl0v/dg/issues
+// @downloadURL  https://github.com/fl0v/dg/raw/master/planets.stats.user.js
+// @match        https://*.darkgalaxy.com/planet/*/production/
+// @match        https://*.darkgalaxy.com/planet/*/training/
+// @match        https://*.darkgalaxy.com/fleet/*
+// @grant        none
+// ==/UserScript==
+
+(function () {
+
+    getInputNumber = function(element) {
+        return getInputOfType(element, 'number');
+    }
+
+    getInputOfType = function(element, type) {
+        if (element) {
+            inputs = element.getElementsByTagName('input');
+            for (let i=0; i<inputs.length; i++) {
+                if (inputs[i].type.toLowerCase() == type) {
+                    return inputs[i];
+                }
+            }
+        }
+        return null;
+    }
+
+    onCheck = function(event) {
+        fillTextForm(event.target.parentNode.parentNode, !event.target.checked);
+    }
+
+    fillTextForm = function(lineDiv, clear) {
+        let input = getInputNumber(lineDiv);
+        if (input) {
+            input.value = clear ? '' : 999999999;
+        }
+    }
+
+    submitForm = function() {
+        HTMLFormElement.prototype.submit.call(document.getElementById('addQueue'));
+    }
+
+    fillTextFormThenSubmit = function(event) {
+        fillTextForm(event.target.parentNode.parentNode);
+        submitForm();
+    }
+
+    getTrainingFormDiv = function(doc) {
+        return (doc ? doc : document).querySelector('form#addQueue .trainList.availableItem.entry');
+    }
+
+    fillTrainingFormThenSubmit = function(event) {
+        fillTextForm(getTrainingFormDiv());
+        submitForm();
+    }
+
+    const parser = new DOMParser();
+    ajaxTrainMultiple = function(event) {
+        fillTextForm(getTrainingFormDiv());
+        let div = event.target.parentNode;
+        let nb = parseInt(div.dataset.nb);
+        ajaxTrainRecursive(div, nb);
+    }
+
+    ajaxTrainRecursive = function(div, nb) {
+        if (nb >= 2 && nb <= 16) {
+            $.ajax({
```

Comment:
did you check with new dg masters ? are we allowed to do ajax request to the engine ?

Reply:
From Discord ![SmartSelect_20230723_001831_Discord](https://github.com/fl0v/dg/assets/16399113/73f35c4c-8dc3-49a5-a65c-2010541b5c5b)

Reply:
cool :)